src/QsCompiler/CompilationManager/Diagnostics.cs

```diff
@@ -82,7 +83,7 @@ private static string Code(QsCompilerDiagnostic msg)
         /// Throws an ArgumentNullException if the Range of the given CompilerDiagnostic is null.
         /// Throws an ArgumentOutOfRangeException if the contained range contains zero or negative entries, or if its Start is bigger than its End.
         /// </summary>
-        internal static Diagnostic Generate(string filename, QsCompilerDiagnostic msg, Position? positionOffset = null)
+        internal static Diagnostic Generate(string filename, QsCompilerDiagnostic msg, [AllowNull] Position? positionOffset = null)
```

Comment:
Same here with `[AllowNull]`. `Position?` should already be a nullable type.

Reply:
Removed